server/hooks.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+// UserHasJoinedTeam is invoked after the membership has been committed to the database. If
+// actor is not nil, the user was added to the team by the actor.
+func (p *Plugin) UserHasJoinedTeam(c *plugin.Context, teamMember *model.TeamMember, actor *model.User) {
+	data := p.constructMessageTemplate(teamMember.UserId, teamMember.TeamId)
+	if data == nil {
+		return
+	}
+
+	for _, message := range p.getWelcomeMessages() {
+		if message.TeamName == data.Team.Name {
+			go p.processWelcomeMessage(*data, *message)
+		}
+	}
+}
+
+// UserHasJoinedChannel is invoked after the membership has been committed to
+// the database. If actor is not nil, the user was invited to the channel by
+// the actor.
+func (p *Plugin) UserHasJoinedChannel(
+	c *plugin.Context,
+	channelMember *model.ChannelMember,
+	_ *model.User,
+) {
```

Comment:
Let's stay consistent with the code style of MM
```suggestion
func (p *Plugin) UserHasJoinedChannel(c *plugin.Context, channelMember *model.ChannelMember, _ *model.User) {
```